src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -19,103 +19,149 @@
 
     SettingsTable(String keyspace)
-    {
-        this(keyspace, DatabaseDescriptor.getRawConfig());
-    }
-
-    SettingsTable(String keyspace, Config config)
     {
         super(TableMetadata.builder(keyspace, "settings")
                            .comment("current settings")
                            .kind(TableMetadata.Kind.VIRTUAL)
                            .partitioner(new LocalPartitioner(UTF8Type.instance))
                            .addPartitionKeyColumn(NAME, UTF8Type.instance)
                            .addRegularColumn(VALUE, UTF8Type.instance)
+                           .addRegularColumn(MUTABLE, BooleanType.instance)
                            .build());
-        this.config = config;
+        for (String name : BACKWARDS_COMPATABLE_NAMES.keySet())
+        {
+            if (DatabaseDescriptor.hasProperty(name))
+            {
+                throw new AssertionError(String.format("Name '%s' is present in Config, this adds a conflict as this " +
+                                                       "name had a different meaning in the '%s",
+                                                       name, SettingsTable.class.getSimpleName()));
+            }
+        }
+    }
+
+    @Override
+    protected void applyColumnUpdate(AbstractMutableVirtualTable.ColumnValues partitionKey,
+                                     AbstractMutableVirtualTable.ColumnValues clusteringColumns,
+                                     Optional<AbstractMutableVirtualTable.ColumnValue> columnValue)
+    {
+        String name = partitionKey.value(0);
+        String value = columnValue.map(v -> v.value().toString()).orElse(null);
+        if (value == null)
+            throw new InvalidRequestException(String.format("Setting the value to null is equivalent to the deletion operation. " +
+                                              "Column deletion is not supported by the '%s.%s'", metadata().keyspace, metadata().name));
+        else
+            runExceptionally(() -> setPropertyFromString(getKeyAndWarnIfObsolete(name), value),
+                             t -> invalidRequest("Invalid update request '%s'. Cause: %s", name, t.getMessage()));
     }
 
     @Override
     public DataSet data(DecoratedKey partitionKey)
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
         String name = UTF8Type.instance.compose(partitionKey.getKey());
-        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
-            ClientWarn.instance.warn("key '" + name + "' is deprecated; should switch to '" + BACKWARDS_COMPATABLE_NAMES.get(name) + "'");
-        if (PROPERTIES.containsKey(name))
-            result.row(name).column(VALUE, getValue(PROPERTIES.get(name)));
+        try
+        {
+            Object value = getPropertyAsString(getKeyAndWarnIfObsolete(name));
+            result.row(name).column(VALUE, value).column(MUTABLE, DatabaseDescriptor.isMutableProperty(name));
```

Comment:
@Mmuzaf I was thinking about this some time ago but I just find some time to go over this again ... I think that `isMutableProperty` is too much here in such sense that it might be "cached". We are not going to introduce a new property while a node is running - it is a static set of keys and values. You can populate a map in a constructor of vtable. So it would be just a simple map lookup, or even easier, name of mutable properties would be a set. So if it is not a set it is not mutable. You do not need to do reflection every single query for every single property all over again.

Reply:
@smiklosovic agree, this makes sense to me. I'll try to figure out the best way to avoid using reflection every time we need to get the mutability state.